xload/load_test.go

```diff
@@ -688,5 +686,18 @@ func runTestcases(t *testing.T, testcases []testcase) {
 			require.NoError(t, err)
 			assert.EqualValues(t, tc.want, tc.input)
 		})
+
+		t.Run("LoadAsync_"+tc.name, func(t *testing.T) {
```

Comment:
And when `env:",required,prefix=SUB_"` is used, error returned is `ErrMissingKey`, not really sure about how this case should behave.

Reply:
Actually this particular testcase should always be an error.

Let me figure out if there’s anything else

Reply:
I’m curious about how a struct can be marked as `required` and hence all its nested fields too (maybe).

And I’d also like to see a testcase for it. Makes it easier to reason about.

Reply:
>  how a struct can be marked as required and hence all its nested fields too (maybe).

Currently, this can be achieved by adding `required` on necessary fields of the struct.

Eg:
```
{
        Name   string `env:"NAME"`
        Age    int    `env:"AGE"`
        Bio    string `env:"BIO"`
        Avatar struct {
            URL    string `env:"URL,required"`
            Height int    `env:"HEIGHT,required"`
            Width  int    `env:"WIDTH,required"`
        } `env:",prefix=AVATAR_"`
    }
```

But, by allowing `required` for the struct, this check becomes implicit. This is easy for native types, but might start becoming complex when nested structs have custom types or more nested structs.

Reply:
Okay, let's keep it explicit.